iOS_SDK/OneSignalSDK/Source/OneSignal.h

```diff
@@ -396,75 +410,69 @@ typedef NS_ENUM(NSUInteger, ONE_S_LOG_LEVEL) {
-+ (id)initWithLaunchOptions:(NSDictionary*)launchOptions appId:(NSString*)appId;
-+ (id)initWithLaunchOptions:(NSDictionary*)launchOptions appId:(NSString*)appId handleNotificationAction:(OSHandleNotificationActionBlock)actionCallback;
-+ (id)initWithLaunchOptions:(NSDictionary*)launchOptions appId:(NSString*)appId handleNotificationAction:(OSHandleNotificationActionBlock)actionCallback settings:(NSDictionary*)settings;
-+ (id)initWithLaunchOptions:(NSDictionary*)launchOptions appId:(NSString*)appId handleNotificationReceived:(OSHandleNotificationReceivedBlock)receivedCallback handleNotificationAction:(OSHandleNotificationActionBlock)actionCallback settings:(NSDictionary*)settings;
++ (void)setAppId:(NSString* _Nonnull)newAppId;
++ (void)setLaunchOptions:(NSDictionary* _Nullable)launchOptions;
++ (void)setAppSettings:(NSDictionary* _Nonnull)settings;
 
 // - Privacy
 + (void)consentGranted:(BOOL)granted;
 + (BOOL)requiresUserPrivacyConsent; // tells your application if privacy consent is still needed from the current user
 + (void)setRequiresUserPrivacyConsent:(BOOL)required; //used by wrapper SDK's to require user privacy consent
 
-@property (class) OSNotificationDisplayType inFocusDisplayType;
+@property (class) OSNotificationDisplayType notificationDisplayType;
 
-+ (NSString*)app_id;
++ (NSString*)appId;
 + (NSString*)sdk_version_raw;
 + (NSString*)sdk_semantic_version;
 
 // Only use if you set kOSSettingsKeyAutoPrompt to false
-+ (void)registerForPushNotifications __deprecated_msg("Please use promptForPushNotificationsWithUserResponse instead.");
 + (void)promptForPushNotificationsWithUserResponse:(void(^)(BOOL accepted))completionHandler;
 + (void)promptForPushNotificationsWithUserResponse:(void (^)(BOOL accepted))completionHandler fallbackToSettings:(BOOL)fallback;
 
-// This method opens the iOS Settings app and navigates to the Push Notification Settings
-// page for your app specifically
-+ (void)presentAppSettings;
 + (void)registerForProvisionalAuthorization:(void(^)(BOOL accepted))completionHandler;
 
+// - Blocks
++ (void)setNotificationWillShowInForegroundHandler:(OSNotificationWillShowInForegroundBlock _Nonnull)block;
++ (void)setNotificationOpenedHandler:(OSNotificationOpenedBlock _Nonnull)block;
++ (void)setInAppMessageClickHandler:(OSInAppMessageClickBlock _Nonnull)block;
+
 // - Logging
 + (void)setLogLevel:(ONE_S_LOG_LEVEL)logLevel visualLevel:(ONE_S_LOG_LEVEL)visualLogLevel;
-+ (void)onesignal_Log:(ONE_S_LOG_LEVEL)logLevel message:(NSString*)message;
++ (void)onesignal_Log:(ONE_S_LOG_LEVEL)logLevel message:(NSString* _Nonnull)message;
 
 // - Tagging
-+ (void)sendTag:(NSString*)key value:(NSString*)value onSuccess:(OSResultSuccessBlock)successBlock onFailure:(OSFailureBlock)failureBlock;
++ (void)sendTag:(NSString* _Nonnull)key value:(NSString* _Nonnull)value onSuccess:(OSResultSuccessBlock _Nullable)successBlock onFailure:(OSFailureBlock _Nullable)failureBlock;
 + (void)sendTag:(NSString*)key value:(NSString*)value;
```

Comment:
`+ (void)sendTag:(NSString*)key value:(NSString*)value;`
should have `_Nonnull`
`+ (void)sendTag:(NSString* _Nonnull)key value:(NSString* _Nonnull)value;`

Reply:
added _nonnull